numpy/core/arrayprint.py

```diff
@@ -28,12 +35,14 @@
     except ImportError:
         from dummy_thread import get_ident
 
+import numpy as np
```

Comment:
You're right, there's definitely precedent for how you had this before. Maybe leave it as it was, and we can deal with `_globals` in general at a later point (it needs the new exception types too).